### ISSUE TITLE
Add support for rel="canonical"

### ISSUE THREAD

**Issue by bugok** [Vimux/Binario#77]
This is a feature request. While this is something which can be implemented by sites using the theme, it would be nice to have this support in the theme itself. 

The idea is to have what [sk3](https://github.com/J-Siu/hugo-theme-sk3/tree/master#site-config) is using in the `enableUserCanonical` parameter. 

I found that this is needed in order to overcome an issue with Google Search Console, where public sites which refer to my site do so by adding a URL parameter, which generates an indexing error in Google Search console. 
See discussion here: https://support.google.com/webmasters/thread/216999759/

The code potential change is minimal, and would require adding more documentation on top of it in order to create a clean PR.
https://gist.github.com/bugok/57addf42a5a47ae8e5594bb8af242386

I'd be happy to submit a PR for this myself, I'm opening this issue to understand if @Vimux thinks this is a good idea or not. If it's a good idea - please let me know how you'd like the PR to look in terms of expected changes, and testing.

Thanks!

**Comment by Vimux** [comment]
I'd be happy to add something like this to the theme, but it looks like it would create more problems than it solves. Using a permalink doesn't seem to work for every page as-is (pagination pages, because `.Permalink` doesn't return the `/page/:number` part, for example). For example, https://example.com/page/2/ would contain `https://example.com/` canonical link. This goes against Google's recommendations for such pages ("*Don't use the first page of a paginated sequence as the canonical page*"). It is an obvious, but probably not the only, edge case in this area.

This example above shows that such things are still best dealt with at the site level. Unfortunately. Otherwise, you could break something else and not even notice. It would be nice to document such things, but I don't have the expertise and free time to cover all such SEO difficulties on one page.

P.S. Google recommends using absolute paths for canonicals rather than relative ones.

**References**:
- https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls#rel-canonical-link-method
- https://developers.google.com/search/docs/specialty/ecommerce/pagination-and-incremental-page-loading#use-urls-correctly

**Comment by Vimux** [comment]
> `enableUserCanonical`

It's a matter of taste, but why not `relCanonical` or something similar?

**Comment by bugok** [comment]
Makes sense. I'm unfamiliar with the details of how this will be applied to multiple pages.

Thanks for pointing me to the references.

From my end, this issue can be closed. I'll try and fix this on my specific site.

**Comment by Vimux** [comment]
I think I'll leave this issue open for now. Perhaps at some point, it will motivate me to consider mentioning such cases somewhere in the docs.
